### PR TITLE
fix watchdog-fencing with topology

### PR DESCRIPTION
**Pull request by wenningerk**
When the node to be fenced via watchdog-fencing doesn't advertise self-fencing
calculation of overall fencing time in a topology doesn't take watchdog-timeout
into account.

### CODE REVIEW

**Comment by wenningerk** [comment]
Seems as if modifying just the calculation of the overall timeout isn't enough as individual actions are observed as well and the operation running into timeout seems not to terminate this observation. Looking more like an independent issue though.

**Comment by wenningerk** [comment]
Yep - that was an independent issue with a dangling timer

**Comment by wenningerk** [comment]
So far it corrects the behavior but we're still getting misleading logs of the following form:

info: All fencing options targeting node4 for client stonith_admin.573557@node2 failed | id=fc3ac6d0
notice: Couldn't find anyone to fence (off) node4 using watchdog 

Making the 2nd watchdog-aware so that it states that target-node couldn't be contacted for a suicide-request and that we thus have to rely that the hardware-watchdog on the target will do it's job (or alike) is easy.
The first is a bit tough and thus questionable if worth the effort.

File: daemons/fenced/fenced_remote.c [hunk 1]
```diff
@@ -1532,6 +1532,18 @@ get_op_total_timeout(const remote_fencing_op_t *op,
                 continue;
             }
             for (device_list = tp->levels[i]; device_list; device_list = device_list->next) {
+                /* in case of watchdog-device we add the timeout to the budget
+                   regardless of if we got a reply or not
+                 */
+                if ((stonith_watchdog_timeout_ms > 0)
+                    && !pcmk__str_eq(op->action, "on", pcmk__str_none)
```
Comment (kgaillot) [inline daemons/fenced/fenced_remote.c:1539]: maybe use pcmk__is_fencing_action() instead

**Review by kgaillot — COMMENTED** [review]
This approach makes sense to me. How is the watchdog timeout added when the target is visible and reported back? Are we adding it twice in that scenario now?

> So far it corrects the behavior but we're still getting misleading logs of the following form:
> 
> info: All fencing options targeting node4 for client stonith_admin.573557@node2 failed | id=fc3ac6d0
> notice: Couldn't find anyone to fence (off) node4 using watchdog
> 
> Making the 2nd watchdog-aware so that it states that target-node couldn't be contacted for a suicide-request and that we thus have to rely that the hardware-watchdog on the target will do it's job (or alike) is easy.

That sounds helpful

> The first is a bit tough and thus questionable if worth the effort.

How about "All non-watchdog fencing options"?

**Comment by wenningerk** [comment]
> This approach makes sense to me. How is the watchdog timeout added when the target is visible and reported back? Are we adding it twice in that scenario now?

That shouldn't happen as for watchdog it more or less just skips searching the query results and adds
the timeout unconditionally.
> 
> 
> How about "All non-watchdog fencing options"?

Yep - probably an easy way out as long as we are missing options to be more context-sensitive which
would probably require more refactoring.
What we could still do is checking if watchdog-fencing is enabled at all and otherwise fall back to
the previous message - just not to confuse people that don't deal with watchdog-fencing at all.

File: cts/support/fence_dummy.in [hunk 2]
```diff
@@ -500,6 +509,11 @@ def main():
         else:
             exitcode = 1
 
+    elif action == "off":
+        if "-F" in options:
+            time.sleep(int(options["-F"]))
```
Comment (kgaillot) [inline cts/support/fence_dummy.in:514]: this will be in addition to any delay from -f -- just making sure that was intentional

Comment (wenningerk) [reply cts/support/fence_dummy.in:514]: Yep that was intentional and on top the easiest least intrusive.
For my purpose I needed everything to finish within timeouts but the actual fencing.
So it is assumed that -f isn't used in combination.

Comment (wenningerk) [reply cts/support/fence_dummy.in:514]: stated that it is an additional wait in description.

**Comment by wenningerk** [comment]
> 
> > How about "All non-watchdog fencing options"?
> 
> Yep - probably an easy way out as long as we are missing options to be more context-sensitive which would probably require more refactoring. What we could still do is checking if watchdog-fencing is enabled at all and otherwise fall back to the previous message - just not to confuse people that don't deal with watchdog-fencing at all.

Guess this is looking way more consistent an easier to understand now:
```
info: All non-watchdog fencing options targeting node4 for client stonith_admin.16553@node2 failed | id=81f5449e
info: Couldn't contact watchdog-fencing target-node (node4)
notice: Waiting 20s for node4 to self-fence (off) for client stonith_admin.16553 | id=81f5449e
```

**Comment by kgaillot** [comment]
nice

**Pull request merged**